server/src/main/java/org/opensearch/indices/recovery/RecoveryTarget.java

```diff
@@ -219,7 +220,15 @@ public void prepareForTranslogOperations(int totalTranslogOps, ActionListener<Vo
 
     @Override
     public void finalizeRecovery(final long globalCheckpoint, final long trimAboveSeqNo, ActionListener<Void> listener) {
-        ActionListener.completeWith(listener, () -> {
+        final StepListener<Void> segrepListener = new StepListener<>();
+        // Force segment refresh on primary relocation which is recovered as replica initially & reset engine
+        if (this.indexShard.indexSettings().isSegRepEnabled() && this.state().getPrimary() == true) {
```

Comment:
`this.state().getPrimary()` - would this start returning true before the relocation handoff is complete?

Reply:
For primary relocation, the `state().getPrimary()` on target would be true as it is primary relocation.